akka-sse/src/main/scala/de/heikoseeberger/akkasse/LineParser.scala

```diff
@@ -54,15 +54,17 @@ private final class LineParser(maxLineSize: Int) extends GraphStage[FlowShape[By
               case ((slices, from, 1), (until, k)) => (slices :+ buffer.slice(from, until).utf8String, until + k, k)
               case ((slices, from, _), _)          => (slices, from, 1)
             }
-          buffer = buffer.drop(nrOfConsumedBytes)
-          lines
+          (buffer.drop(nrOfConsumedBytes), lines)
+        }
+
+        parseLines(buffer ++ grab(in)) match {
+          case (buf, _) if buf.size > maxLineSize =>
+            buffer = ByteString.empty // No need to retain this buffer
```

Comment:
Is this needed in the case of failure (`failStage`)? Ain't the whole stage disposed?